tests/testapp/tests/test_filter_classes.py

```diff
@@ -177,3 +178,23 @@ def test_browsable_api_filter_member(self):
         self.assertContains(response, 'test-shelf-a</option>')
         self.assertNotContains(response, 'org_b</option>')
         self.assertNotContains(response, 'test-shelf-b</option>')
+
+    def test_unauthorized_user(self):
+        r = self.client.get(reverse("test_shelf_list_unauthorized_view"))
+        self.assertEqual(r.status_code, 401)
+        r = self.client.get(
+            reverse("test_book_list_unauthorized_view", args=(self.shelf_a.id,))
+        )
+        self.assertEqual(r.status_code, 401)
+
+    def test_unauthorized_user_failing(self):
+        del FilterByOrganization.permission_classes
+        del FilterByParent.permission_classes
```

Comment:
I think I didn't explain myself properly. This is not required.
The test above is enough, we just have to ensure it fails without the changes to the code in the PR.

I tried this and it works, here's how to do it in this case:

- comment out `permission_classes` in the filter classes
- run tests
- expect the test above to fail with "AttributeError: 'AnonymousUser' object has no attribute 'organizations_dict'"

If it fails, it means the test is correct, if it doesn't it mean the test is missing something.

I confirm the test above is correct, this one below is not necessary and I'm removing it.